### PR TITLE
fix #751 [rule.add-constant] add ignoreFuncs to exclude constants in …

### LINKED ISSUE #751: [rule.add-constant] add ''ignoreFuncs' to exclude constants in functions
**Is your feature request related to a problem? Please describe.**
I want to add 'ignoreFuncs' in configuration
to have a possibility to exclude constants in functions

```
[rule.add-constant]
  arguments = [{'ignoreFuncs="println,os.Chmod"}]
```

**Describe the solution you'd like**
I want to create PR with changes, which add this feature

### PR DESCRIPTION
**Pull request by rawen17**
Closes #751

### CODE REVIEW

File: rule/add-constant.go [hunk 1]
```diff
@@ -158,6 +214,18 @@ func (r *AddConstantRule) configure(arguments lint.Arguments) {
 						panic(fmt.Sprintf("Invalid argument to the add-constant rule, expecting string representation of an integer. Got '%v'", v))
 					}
 					r.strLitLimit = limit
+				case "ignoreFuncs":
+					excludes, ok := v.(string)
+					if !ok {
+						panic(fmt.Sprintf("Invalid argument to the add-constant rule, string expected. Got '%v' (%T)", v, v))
+					}
+
+					for _, exclude := range strings.Split(excludes, ",") {
```
Comment (chavacava) [inline rule/add-constant.go:223]: trimming blanks can help in avoiding subtle config problems.
For example configuring the rule with 
```go
"os.*,.Println,ignoredFunc"
```
will not behave as if it is configured with
```go
"os.*, .Println,ignoredFunc"
```

**Review by chavacava — CHANGES_REQUESTED** [review]
I've left some comments
Please fill the PR description by using the PR description template

In general I think regexps are too powerful (or complicated?) to be used without the risk of ignoring things you don't know you are ignoring. For example, the regexp `.Println` used in the tests will not ignore just "methods named `Println` of any package" but **also** any method or function named such as `Println` is a suffix of the name (`myPrintln`, `ThisIsAnotherPrintln`, `dPrintlln`, ...)

**Comment by chavacava** [comment]
My point is that RE are very error prone (at the point that your own test case `.Println` was wrong)
I'm not against the use of RE and I agree that providing clear doc will help in avoiding subtle errors; for example in the rule documentation it should be interesting to highlight that `fmt.Println` in the given example, will make `revive` ignore `fmt.Println`, `fmtXPrintln`, `fmt1Println`, an so on.

**Review by chavacava — APPROVED** [review]

**Comment by chavacava** [comment]
> It was my fault, i had to write regexp `os\\.*,fmt\\.Println,make` in documentation, but i wrote `os.*,fmt.Println,make ` I will fix it

@rawen17 thanks for the PR, I've already merged it. 
Sorry @xobotyi I did not notice that I was answering to you (and not to the author of the PR)

**Comment by rawen17** [comment]
@chavacava Ok, thanks, i will create new PR for this changes

**Pull request merged**